test/jdk/java/io/File/EmptyPath.java

```diff
@@ -22,28 +22,278 @@
  */
 
 /* @test
-   @bug 4842706
-   @summary Test some file operations with empty path
+ * @bug 4842706 8024695
+ * @summary Test some file operations with empty path
+ * @run junit EmptyPath
  */
 
-import java.io.*;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.FileStore;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestMethodOrder(OrderAnnotation.class)
 public class EmptyPath {
-    public static void main(String [] args) throws Exception {
-        File f = new File("");
-        f.mkdir();
-        try {
-            f.createNewFile();
-            throw new RuntimeException("Expected exception not thrown");
-        } catch (IOException ioe) {
-            // Correct result
-        }
-        try {
-            FileInputStream fis = new FileInputStream(f);
-            fis.close();
-            throw new RuntimeException("Expected exception not thrown");
-        } catch (FileNotFoundException fnfe) {
-            // Correct result
+    private static final String EMPTY_STRING = "";
+
+    static File f;
+    static Path p;
+
+    @BeforeAll
+    public static void init() {
+        f = new File(EMPTY_STRING);
+        p = Path.of(EMPTY_STRING);
+    }
+
+    @Test
+    @Order(1)
```

Comment:
What is the reason for use `@Order` in these tests, is there a dependency? Asking because we've usually avoided need to do this and I'm trying to see if how someone might be able to add to this test in the future.

Reply:
Early on I had some concern about the state being affected in a way that would make some tests spuriously fail if they were not performed before others. I think with careful handling of resetting the state as suggested above ("Need try-finally in the setter tests") then the `Order` annotations can be removed.